File: benchmarks/js/dto.js
class User{ constructor(id,name,age,isActive,score){ this.id=id;this.name=name;this.age=age;this.isActive=isActive;this.score=score; } }
class UserDTO{ constructor(id,displayName,score){ this.id=id;this.displayName=displayName;this.score=score; } }
function compute(){
  const count=100000; const users=[];
  for(let i=0;i<count;i++) users.push(new User(i,"User_"+i,(i%60)+10,(i%2)===0,(i%100)*1.5));
  const aa=[]; let ts=0.0;
  for(let i=0;i<count;i++){ const u=users[i]; if(u.isActive&&u.age>=18){ aa.push(new UserDTO(u.id,u.name,u.score)); ts=ts+u.score; } }
  return aa.length+ts;
}
for(let w=0;w<3;w++) compute();
let best=Infinity, chk;
for(let r=0;r<10;r++){ const t=performance.now(); chk=compute(); const ms=performance.now()-t; if(ms<best)best=ms; }
console.error(chk); console.log(best.toFixed(3));

File: benchmarks/py/dto.py
from _harness import run


class User:
    __slots__ = ("id", "name", "age", "isActive", "score")

    def __init__(self, id, name, age, isActive, score):
        self.id = id
        self.name = name
        self.age = age
        self.isActive = isActive
        self.score = score


class UserDTO:
    __slots__ = ("id", "displayName", "score")

    def __init__(self, id, displayName, score):
        self.id = id
        self.displayName = displayName
        self.score = score


def compute():
    count = 100000
    users = []
    for i in range(count):
        users.append(User(i, "User_" + str(i), (i % 60) + 10, (i % 2) == 0, (i % 100) * 1.5))
    aa = []
    ts = 0.0
    for i in range(count):
        u = users[i]
        if u.isActive and u.age >= 18:
            aa.append(UserDTO(u.id, u.name, u.score))
            ts = ts + u.score
    return len(aa) + ts


run(compute)
